Makefile.proto

```diff
@@ -13,60 +13,118 @@
+PROTOC_VERSION := 3.14.0
 
-default: protobuf
+TOOLS_DIR                       := $(abspath ./.tools)
+TOOLS_MOD_DIR                   := ./internal/tools
+PROTOBUF_VERSION                := v1
+OTEL_PROTO_SUBMODULE            := exporters/otlp/internal/opentelemetry-proto
+GEN_TEMP_DIR                    := gen
+SUBMODULE_PROTO_FILES           := $(wildcard $(OTEL_PROTO_SUBMODULE)/opentelemetry/proto/*/$(PROTOBUF_VERSION)/*.proto) $(wildcard $(OTEL_PROTO_SUBMODULE)/opentelemetry/proto/collector/*/$(PROTOBUF_VERSION)/*.proto)
 
-.PHONY: protobuf protobuf-source gen-protobuf copy-protobufs
-protobuf: protobuf-source gen-protobuf copy-protobufs
+ifeq ($(strip $(SUBMODULE_PROTO_FILES)),)
+$(error Submodule at $(OTEL_PROTO_SUBMODULE) is not checked out, use "git submodule update --init")
+endif
+
+PROTOBUF_GEN_DIR   := exporters/otlp/internal/opentelemetry-proto-gen
+PROTOBUF_TEMP_DIR  := $(GEN_TEMP_DIR)/pb-go
+PROTO_SOURCE_DIR   := $(GEN_TEMP_DIR)/proto
+SOURCE_PROTO_FILES := $(subst $(OTEL_PROTO_SUBMODULE),$(PROTO_SOURCE_DIR),$(SUBMODULE_PROTO_FILES))
 
-protobuf-source: $(SOURCE_PROTO_FILES) | $(PROTO_SOURCE_DIR)/
+.DEFAULT_GOAL := protobuf
 
-# Changes go_package in .proto file to point to repo-local location
-define exec-replace-pkgname
-sed  's,go_package = "github.com/open-telemetry/opentelemetry-proto/gen/go,go_package = "go.opentelemetry.io/otel/exporters/otlp/internal/opentelemetry-proto-gen,' < $(1) > $(2)
+UNAME_S := $(shell uname -s)
+UNAME_M := $(shell uname -m)
 
+# Returns a URL to protoc zip archive.
+#
+# Example: $(call get-protoc-zip-url,linux,x86_64)
+define get-protoc-zip-url
+https://github.com/protocolbuffers/protobuf/releases/download/v$(PROTOC_VERSION)/protoc-$(PROTOC_VERSION)-$(1)-$(2).zip
 endef
 
-# replace opentelemetry-proto package name by go.opentelemetry.io/otel specific version
-$(SOURCE_PROTO_FILES): $(PROTO_SOURCE_DIR)/%.proto: $(OTEL_PROTO_SUBMODULE)/%.proto
-	@mkdir -p $(@D)
-	$(call exec-replace-pkgname,$<,$@)
+$(TOOLS_DIR)/PROTOC_$(PROTOC_VERSION):
+	@touch "$@"
+
+# Depend on a versioned file (like PROTOC_3.14.0), so when version
+# gets bumped, we will depend on a nonexistent file and thus download
+# a newer version.
+$(TOOLS_DIR)/protoc/bin/protoc: $(TOOLS_DIR)/PROTOC_$(PROTOC_VERSION)
+	echo "Fetching protoc $(PROTOC_VERSION)"
+	rm -rf $(TOOLS_DIR)/protoc
+ifeq ($(UNAME_S),Linux)
+	wget -O $(TOOLS_DIR)/protoc.zip $(call get-protoc-zip-url,linux,$(UNAME_M)) && \
+	unzip $(TOOLS_DIR)/protoc.zip -d $(TOOLS_DIR)/protoc-tmp && \
+	rm $(TOOLS_DIR)/protoc.zip && \
+	touch $(TOOLS_DIR)/protoc-tmp/bin/protoc && \
+	mv $(TOOLS_DIR)/protoc-tmp $(TOOLS_DIR)/protoc
+else ifeq ($(UNAME_S),Darwin)
+	# TODO(krnowak): I have no idea if this works, not a mac user
+	# here.
+	wget -O $(TOOLS_DIR)/protoc.zip $(call get-protoc-zip-url,osx,x86_64) && \
+	unzip $(TOOLS_DIR)/protoc.zip -d $(TOOLS_DIR)/protoc-tmp &&
+	rm $(TOOLS_DIR)/protoc.zip &&
+	mv $(TOOLS_DIR)/protoc-tmp $(TOOLS_DIR)/protoc
```

Comment:
```suggestion
	unzip $(TOOLS_DIR)/protoc.zip -d $(TOOLS_DIR)/protoc-tmp && \
	rm $(TOOLS_DIR)/protoc.zip && \
	mv $(TOOLS_DIR)/protoc-tmp $(TOOLS_DIR)/protoc
```

Reply:
Thanks for testing, I appreciate that. Since code paths for linux and darwin were mostly the same (they only differ in URL for protoc zip archive), I simplified the rule for fetching protoc.